### PR TITLE
API Review: cookieManager for multiprofile

### PR DESCRIPTION
**Pull request by zhuhaichao518**
Added interface to get cookie manager from profile.

### CODE REVIEW

File: specs/MultiProfile.md [hunk 1]
```diff
@@ -296,6 +306,8 @@ namespace Microsoft.Web.WebView2.Core
         Boolean IsInPrivateModeEnabled { get; };
 
         String ProfilePath { get; };
+
+        CoreWebView2CookieManager cookieManager { get; };
```
Comment (david-risney) [inline specs/MultiProfile.md:310]: This should be capital **C**ookieManager. If this was generated for you by build-apiwriter.ps1, then you may need to add CookieManager to the list under `# capitalize the first letter for properties` in buid-apiwriter.ps1.

```suggestion
        CoreWebView2CookieManager CookieManager { get; };
```

File: specs/MultiProfile.md [hunk 2]
```diff
@@ -107,6 +108,9 @@ HRESULT AppWindow::OnCreateCoreWebView2ControllerCompleted(HRESULT result, ICore
         BOOL inPrivateModeEnabled = FALSE;
         CHECK_FAILURE(profile->get_IsInPrivateModeEnabled(&inPrivateModeEnabled));
         CHECK_FAILURE(profile->get_ProfileName(&m_profileName));
+
+        // Gets the cookieManager which can be used the same as described in CookieManagement.md
+        CHECK_FAILURE(profile->get_CookieManager(&m_cookieManager));
```
Comment (david-risney) [inline specs/MultiProfile.md:113]: You need to add new samples. This sample is using the profile name and InPrivate status to update the title and icon of the window. The CookieManager isn't going to help with updating the title and icon of the window. For a new sample you may want to look at the CookieManager API spec and adapt its sample code.

File: specs/MultiProfile.md [hunk 3]
```diff
@@ -245,6 +252,9 @@ interface ICoreWebView2Profile : IUnknown {
   /// Full path of the profile directory.
   [propget] HRESULT ProfilePath([out, retval] LPWSTR* value);
 
+  /// Get the cookie manager object for the profile.
+  [propget] HRESULT CookieManager([out, retval] ICoreWebView2CookieManager** cookieManager);
```
Comment (david-risney) [inline specs/MultiProfile.md:256]: ICoreWebView2Profile has already been released as experimental. So we won't be updating the existing ICoreWebView2Profile interface but rather making a new interface named ICoreWebView2Profile2. Although the experimental and staging versions don't derive from each other the public versions will and what we show in our API specs is the expected public interface.
```
interface ICoreWebView2Profile2 : ICoreWebView2Profile {
  /// Get the cookie manager object for the profile.
  [propget] HRESULT CookieManager([out, retval] ICoreWebView2CookieManager** cookieManager);
}
```

File: specs/MultiProfile.md (hunk 1 above)
Comment (david-risney) [inline specs/MultiProfile.md:310]: After making the change to the COM interface to have CookieManager in a new COM interface, you should see a similar change in this generated MIDL3 with CookieManager in an interface_name block within the CoreWebView2Profile runtimeclass:
```
        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Profile2")]
        {
            CoreWebView2CookieManager CookieManager { get; };
        }
```

**Review by david-risney — CHANGES_REQUESTED** [review]

File: specs/MultiProfile.md (hunk 2 above)
Comment (zhuhaichao518) [reply specs/MultiProfile.md]: Added new samples.

File: specs/MultiProfile.md (hunk 3 above)
Comment (zhuhaichao518) [reply specs/MultiProfile.md:313]: Thanks for the explanation. Added _ICoreWebView2Profile2_ interface.

File: specs/MultiProfile.md (hunk 1 above)
Comment (zhuhaichao518) [reply specs/MultiProfile.md]: Hi, I am not familiar with MIDL3 generation. Could you please tell me how to get it?

File: specs/MultiProfile.md [hunk 4]
```diff
@@ -118,6 +124,49 @@ HRESULT AppWindow::OnCreateCoreWebView2ControllerCompleted(HRESULT result, ICore
...
+{
+    wil::com_ptr<ICoreWebView2> coreWebView2;
+    CHECK_FAILURE(controller->get_CoreWebView2(&coreWebView2));
+    auto webview7 = coreWebView2.try_query<ICoreWebView2_7>();
+    if (webview7)
+    {
+        wil::com_ptr<ICoreWebView2Profile> profile;
+        CHECK_FAILURE(webview7->get_Profile(&profile));
+        auto profile2 = profile.try_query<ICoreWebView2Profile2>;
+        if (profile2){
```
Comment (david-risney) [inline specs/MultiProfile.md:142]:
```suggestion
        if (profile2)
        {
```

File: specs/MultiProfile.md [hunk 5]
```diff
@@ -118,6 +124,49 @@ HRESULT AppWindow::OnCreateCoreWebView2ControllerCompleted(HRESULT result, ICore
...
+        //! [CookieManagerProfile]
```
Comment (david-risney) [inline specs/MultiProfile.md:143]: You can remove the //! comments from this API spec. But for the actual sample code, just one line of calling get_CookieManager is not very helpful. You can put the //! around this whole ScenarioCookieManagement::ScenarioCookieManagement function.

Comment (david-risney) [reply specs/MultiProfile.md:143]:
```suggestion
```

**Review by david-risney — CHANGES_REQUESTED** [review]
Please update the few issues remaining and then please complete the PR, and open a new PR to master and let me know and I will schedule a review. Thanks!

**Pull request merged**